docs(solr-drupal): [SITE-5999] add Dense Vector (AI semantic) search guide for Drupal (#10138)
pantheon-systems/documentation: src/source/content/guides/solr-drupal/05-dense-vector.md

---
title: Apache Solr for Drupal
subtitle: Dense Vector (AI Semantic) Search
description: Set up dense vector (AI-powered semantic) search for Drupal on Pantheon using Solr 9's DenseVectorField and the Search API Solr Dense Vector module.
contenttype: [guide]
innav: [false]
categories: [search]
cms: [drupal10, drupal11]
audience: [development]
product: [search]
integration: [--]
tags: [solr, search, modules]

> contributors: [mehta-asim]

Reviewer: ```suggestion
contributors: [mehta-asim,jazzsequence]
```

---
title: Apache Solr for Drupal
subtitle: Dense Vector (AI Semantic) Search
description: Set up dense vector (AI-powered semantic) search for Drupal on Pantheon using Solr 9's DenseVectorField and the Search API Solr Dense Vector module.
contenttype: [guide]
innav: [false]
categories: [search]
cms: [drupal10, drupal11]
audience: [development]
product: [search]
integration: [--]
tags: [solr, search, modules]

> contributors: [mehta-asim,jazzsequence]

Reviewer: ```suggestion
contributors: [mehta-asim,jazzsequence,rkunjappan]
```

---
title: Apache Solr for Drupal
subtitle: Dense Vector (AI Semantic) Search
description: Set up dense vector (AI-powered semantic) search for Drupal on Pantheon using Solr 9's DenseVectorField and the Search API Solr Dense Vector module.
contenttype: [guide]
innav: [false]
categories: [search]
cms: [drupal10, drupal11]
audience: [development]
product: [search]
integration: [--]
tags: [solr, search, modules]
contributors: [mehta-asim]
reviewed: "2026-06-25"
showtoc: true
permalink: docs/guides/pantheon-search/solr-drupal/dense-vector
editpath: solr-drupal/05-dense-vector.md
---

> This guide covers the dense-vector-specific setup for semantic (AI-powered) search on a Drupal site on Pantheon, using Solr 9's DenseVectorField and the Search API Solr Dense Vector module. It assumes Pantheon Search is already set up; base Solr steps link out to [Apache Solr for Drupal](/guides/pantheon-search/solr-drupal/solr-drupal).

Reviewer: ```suggestion
This guide covers the dense-vector-specific setup for semantic (AI-powered) search on a Drupal site on Pantheon, using Solr 9's `DenseVectorField` and the Search API Solr Dense Vector module. It assumes Pantheon Search is already set up; base Solr steps link out to [Apache Solr for Drupal](/guides/pantheon-search/solr-drupal/solr-drupal).
```

Traditional keyword search matches exact words. Dense vector search converts text into numerical vectors (embeddings) with an AI model, then finds content that is semantically similar. For example, a search for "fixing automobiles" matches an article about "car engine maintenance" even though the two share no keywords.

<Alert title="Beta Release" type="info">

> Dense vector search on Pantheon requires Solr 9.6 or higher (Pantheon runs 9.10.0), and its setup ordering is sensitive. Follow the steps in the exact order described in this guide. Tested on Solr 9.10.0, Drupal 10.6.7 / 11.3.8.

Reviewer 2: ```suggestion
Dense vector search requires Solr 9.6 or higher (Pantheon runs 9.10.1), and its setup ordering is sensitive. Follow the steps in the exact order described in this guide. Tested on Solr 9.10.1, Drupal 10.6.7 / 11.3.8.
```

This setup relies on **alpha** contributed modules: `search_api_solr_dense_vector` (1.0.0-alpha9) for dense vector search, and `ai_search` (2.0.0-alpha2), which is required for RAG search (Step 11). Alpha modules are unstable and can change or break between releases. Test thoroughly on non-production environments and do not rely on them for production search.

This setup uses contributed modules that Pantheon does not support, and an AI embedding provider that you bring yourself. It is not a Pantheon-supported feature.

</Alert>

## Before You Begin

> Complete the base Pantheon Search setup first. This guide does not repeat it:

Reviewer: ```suggestion
- Complete the base [Pantheon Search setup](/guides/pantheon-search/solr-drupal/solr-drupal#before-you-begin) first.
```

> 

Reviewer: ```suggestion
```

- Install and enable the Search API Pantheon module. Refer to [Install the Search API Pantheon Module](/guides/pantheon-search/solr-drupal/solr-drupal#install-the-search-api-pantheon-module).

> - Set Solr 9 in `pantheon.yml` (requires Solr 9.6+; Pantheon runs 9.10.0). Refer to [Configure the Solr Version](/guides/pantheon-search/solr-drupal/solr-drupal#configure-the-solr-version).

Reviewer 2: ```suggestion
- Set Solr 9 in `pantheon.yml` (requires Solr 9.6+; Pantheon runs 9.10.1). Refer to [Configure the Solr Version](/guides/pantheon-search/solr-drupal/solr-drupal#configure-the-solr-version).
```

- Create a Search API server and index with Title and Body fields. Refer to [Add Search Index](/guides/pantheon-search/solr-drupal/solr-drupal#add-search-index).
- Install and authenticate [Terminus](/terminus).
- Make Composer changes with [Integrated Composer](/guides/integrated-composer) (commit only `composer.json` and `composer.lock`).
- Obtain an AI embedding provider account: Google Vertex AI, OpenAI, or Google Gemini (see below).

<Alert title="Note" type="info">

> **Multidev Solr 9 inheritance.** A multidev created from a Dev environment on Solr 8 inherits Solr 8 even when `pantheon.yml` specifies version 9, because the change is not detected as new. To force the upgrade, set the version to 8, commit and push, then set it to 9 and push again.

Reviewer 2: @mehta-asim Is this note correct?

Author: yes but its worded incorrectly and only applied to a certain scenario as follows, inorder to change the solr version from 8 to 9
- i create a local branch from main
- update solr version to 9 in pantheon.yml
- push it up and create a multidev from the branch
- if the dev version has solr 8 the multidev creation will also have 8
- it will ignore the new solr version 9 set in patheon.yml

I have added a suggestion at the bottom for the updated wording

> **Multidev Solr 9 inheritance.** A multidev created from a Dev environment on Solr 8 inherits Solr 8 even when `pantheon.yml` specifies version 9, because the change is not detected as new. To force the upgrade, set the version to 8, commit and push, then set it to 9 and push again.

Reviewer: ```suggestion
**Multidev Solr version inheritance.** A multidev created from a Dev environment on Solr 8 inherits Solr 8 even when `pantheon.yml` specifies version 9, because the change is not detected as new. To force the upgrade, set the version to 8, commit and push, then set it to 9 and push again.
```
